src/Database/Type/EnumType.php

```diff
@@ -0,0 +1,107 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database\Type;
+
+use BackedEnum;
+use Cake\Database\Driver;
+use InvalidArgumentException;
+use PDO;
+
+/**
+ * Enum type converter.
+ *
+ * Use to convert string data between PHP and the database types.
+ */
+class EnumType extends BaseType implements OptionalConvertInterface
```

Comment:
We're pretty close to being able to do all the work in a type class. Have we considered having a userfacing interface like:

```php
// In a table initialize()
$this->setColumnType('status', EnumType::for(ArticleStatus::class));
```

Under the hood we could use `TypeFactory::set()` to create a type instance for each enum type.

The benefits of this approach would be that we wouldn't need a behavior to do mapping in and out of the enum value. We also wouldn't need the additional `instanceof BackedEnum` conditions as they type system would handle conversions like we do with DateTime objects.

Reply:
Let me try...